File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Header/MolTitleSubtitle.swift
import SwiftUI

public struct MolTitleSubtitle: View {
    private let title: String
    private let subtitle: String

    public init(title: String, subtitle: String) {
        self.title = title
        self.subtitle = subtitle
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: SpacingRoles.Component.extraCompact.rawValue) {
            AtomText(text: title, textFontRole: .displayH1, colorRole: .onScrim, maxLines: 1)
            AtomText(text: subtitle, textFontRole: .subheadP2, colorRole: .onScrimVariant, maxLines: 2)
        }
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/header/MolTitleSubtitle.kt
package com.perrystreet.woof.designsystem.atomic.molecules.header

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.SpacingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.text.AtomText
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextColorRole
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextFontRole

@Composable
fun MolTitleSubtitle(
    title: String,
    subtitle: String,
) {
    Column(verticalArrangement = Arrangement.spacedBy(SpacingRoles.Component.ExtraCompact.dp)) {
        AtomText(
            text = title,
            textFontRole = TextFontRole.DisplayH1,
            colorRole = TextColorRole.OnScrim,
            maxLines = 1,
        )
        AtomText(
            text = subtitle,
            textFontRole = TextFontRole.SubheadP2,
            colorRole = TextColorRole.OnScrimVariant,
            maxLines = 2,
        )
    }
}
